File: app/src/main/java/com/malenalbc/superherotest/data/model/ResultWrapper.java

package com.malenalbc.superherotest.data.model;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class ResultWrapper<D extends Result> {

    @SerializedName("code")
    @Expose
    public int code;
    @SerializedName("status")
    @Expose
    public String status;
    @SerializedName("copyright")
    @Expose
    public String copyright;
    @SerializedName("attributionText")
    @Expose
    public String attributionText;
    @SerializedName("attributionHTML")
    @Expose
    public String attributionHTML;
    @SerializedName("etag")
    @Expose
    public String etag;
    @SerializedName("data")
    @Expose
    public ResultContainer<D> data;


    public ResultContainer<D> getData () {
        return data;
    }
}

File: data/src/main/java/com/malenalbc/stormcomics/data/model/ResultWrapper.kt
package com.malenalbc.stormcomics.data.model

import com.google.gson.annotations.SerializedName


class ResultWrapper<D : BaseResult>(
    @SerializedName("code") var code: Int = 0,
    @SerializedName("status") var status: String? = null,
    @SerializedName("copyright") var copyright: String? = null,
    @SerializedName("attributionText") var attributionText: String? = null,
    @SerializedName("attributionHTML") var attributionHTML: String? = null,
    @SerializedName("etag") var etag: String? = null,
    @SerializedName("data") var data: ResultContainer<D>? = null
)
